### ISSUE TITLE
Configs being merged with unwanted original frameworks config files

### ISSUE THREAD

**Issue by MrWeb** [laravel/framework#52539]
### Laravel Version

11.21.0

### PHP Version

8.2.22

### Description

I'm in the situation where in my `config/database.php` I removed some connections like `sqlite` as I don't need them and I need to read the connections via `config('database.connections')`.

Before reading the connections I run `php artisan config:clear` and `php artisan config:cache` at that point I output the contents of the config with `config('database.connections')` and the result I have still has `sqlite`, `postgress` and all other connections I removed, but merged with the extra connections I created.

I tried to rename `config/database.php` to `config/__database.php` to see what error would occur, but no error comes up when doing `config('database.connections')`, I still get the original config as it was prior to my changes. 

I went to `vendor/laravel/framework/config/database.php` and removed `sqlite` from there as a test, I cached the config, and this time the output was without `sqlite` (but with other keys merged).

So basically Laravel is merging my config with the original framework config, this could look like a feature but it annoying as I should be able to remove stuff from it and have them not being present.

### Steps To Reproduce

1. Remove `sqlite` connection from `config/database.php`
2. `php artisan config:clear`
3. `php artisan config:cache`
4. output `config('database.connections')`

You should still see `sqlite` in the connections.
